src/main/java/org/jabref/gui/bibtexextractor/ExtractBibtexDialog.java

```diff
@@ -0,0 +1,72 @@
+package org.jabref.gui.bibtexextractor;
+
+import javafx.scene.control.*;
+import javafx.scene.layout.VBox;
+import org.jabref.Globals;
+import org.jabref.gui.JabRefFrame;
+import org.jabref.gui.util.BaseDialog;
+import org.jabref.logic.l10n.Localization;
+import org.jabref.model.entry.BibEntry;
+import org.jabref.model.entry.BiblatexEntryTypes;
+import org.jabref.model.entry.EntryType;
+
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * GUI Dialog for the feature "Extract BibTeX from plain text".
+ */
+public class ExtractBibtexDialog extends BaseDialog<Void> {
+
+    private final JabRefFrame frame;
+    private TextArea textArea;
+    private Button buttonExtract;
+
+    public ExtractBibtexDialog(JabRefFrame frame) {
```

Comment:
We don't JabRef Frame here.
Best is to use the StateManager and then get the active Database and call insertEntries there